tests/test_sockets.py

```diff
@@ -735,12 +735,17 @@ async def test_bind_link_local(self) -> None:
     sys.platform == "win32", reason="UNIX sockets are not available on Windows"
 )
 class TestUNIXStream:
-    @pytest.fixture
-    def socket_path(self) -> Generator[Path, None, None]:
+    @pytest.fixture(params=["path", "abstract"])
+    def socket_path(
+        self, request: SubRequest, tmp_path_factory: TempPathFactory
+    ) -> Generator[Path, None, None]:
         # Use stdlib tempdir generation
         # Fixes `OSError: AF_UNIX path too long` from pytest generated temp_path
         with tempfile.TemporaryDirectory() as path:
-            yield Path(path) / "socket"
+            if request.param == "path":
+                yield Path(path) / "socket"
+            elif request.param == "abstract":
```

Comment:
Redudant `elif` – what else could `request.param be?

Reply:
Just a remnant from me normally being paranoid with this pattern and a guarding else with assert but I'll just change to an else here.